Visualization/bokeh_holoviews_datashader.ipynb

```diff
@@ -0,0 +1,560 @@
+    "Next we add callback functionality to the plot above and retrieve the indices of the selected points. First, use the box selection tool to create a selection box for the two-dimensional histogram above. Then run the cell below to count the number of datapoints within the selection region."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "selection = (points.data.x > box.bounds[0]) \\\n",
+    "    & (points.data.y > box.bounds[1]) \\\n",
+    "    & (points.data.x < box.bounds[2]) \\\n",
+    "    & (points.data.y < box.bounds[3])\n",
+    "print('The selection box contains %i datapoints'%(np.sum(selection)))\n",
+    "if np.sum(selection) > 0:\n",
+    "    print('\\nHere are some of the selected indices...')\n",
+    "    print(np.nonzero(selection)[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Another option is to make a second linked plot paired with the box selection on the two-dimensional histogram."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First, create a holoviews dataset instance. Here we label some of the columns.\n",
+    "kdims = [('ra', 'RA(deg)'), ('dec', 'Dec(deg)')]\n",
+    "vdims = [('phot_g_mean_mag', 'G(mag)')]\n",
+    "ds = hv.Dataset(df, kdims, vdims)\n",
+    "ds"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "points = hv.Points(ds)\n",
+    "\n",
+    "#boundsxy = (0, 0, 0, 0)\n",
+    "boundsxy = (np.min(ds.data['ra']), np.min(ds.data['dec']), np.max(ds.data['ra']), np.max(ds.data['dec']))\n",
+    "box = streams.BoundsXY(source=points, bounds=boundsxy)\n",
+    "box_plot = hv.DynamicMap(lambda bounds: hv.Bounds(bounds), streams=[box])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This function defines the custom callback functionality to update the linked histogram\n",
+    "def update_histogram(bounds=bounds):\n",
+    "    \n",
+    "    selection = (ds.data['ra'] > bounds[0]) & \\\n",
+    "                (ds.data['dec'] > bounds[1]) & \\\n",
+    "                (ds.data['ra'] < bounds[2]) & \\\n",
+    "                (ds.data['dec'] < bounds[3])\n",
+    "    \n",
+    "    selected_mag = ds.data.loc[selection]['phot_g_mean_mag']\n",
+    "    \n",
+    "    frequencies, edges = np.histogram(selected_mag)\n",
+    "    \n",
+    "    hist = hv.Histogram((np.log(frequencies), edges))\n",
+    "    return hist"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%output size=150\n",
+    "dmap = hv.DynamicMap(update_histogram, streams=[box])\n",
+    "datashade(points, cmap=bokeh.palettes.Viridis256) * box_plot + dmap"
+   ]
```

Comment:
Is there a way to dynamically update the histogram binning?

Reply:
Not sure. It would be a nice feature.